src/test/clojure/xtdb/compactor_test.clj

```diff
@@ -630,3 +630,22 @@
   (c/compact-all! tu/*node*)
 
   (t/is (= [] (xt/q tu/*node* "SELECT _id FROM foo"))))
+
+
+(t/deftest test-same-entity-same-transaction-4303
```

Comment:
@jarohen  The bad thing about this test is that I needed to add a `reverse` to `trie-catalog/current-tries` to make most of these fail in the non fixed version. So it's no really testing much in this case as the natural order of the `current-tries` gives the same splitting behavior.

Reply:
I'm assuming you've cherry-picked this against your TPC-H failure on your other branch? if so, once your other branch is merged too we'll have a regression test for this one